File: Library/film.cpp
#include<string>
#include<iostream>
#include"film.h"

Film::Film(string call_no, string title, string subjects, string director, string notes, string year):Media(call_no,title,subjects,notes),m_director(director),m_year(year)
{
}

void Film::display() const
{
    Media::display();
    cout<<"Director: "<<m_director;
    cout<<"Year: "<<m_year << endl;
}
bool Film::comp_other(const string& str) const
{
    if(Media::m_notes.find(str) != string::npos ||m_director.find(str) != string::npos || m_year.find(str) != string::npos)
    {
        return true;
    }
    else
    {
        return false;
    }
}

File: Library_python /Film.py
from Media import Media

class Film(Media):
   def __init__(self, call_no, title, subjects, director, notes, year):
      super().__init__(call_no, title, subjects)
      self.director = director
      self.year = year
      self.notes = notes

   def display(self):
      super().display()
      print ( "Director: ", self.director, "Notes: ", self.notes, "Year: ", self.year)
   
   def compareByOther(self, search):
      if search in self.director or \
         search in self.year or \
         search in self.notes:
         return True
      else:
         return False

